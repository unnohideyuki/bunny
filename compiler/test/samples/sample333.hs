x :: Double
x = pi

y :: Float
y = pi

main = do print $ floatRadix x
          print $ floatRadix y
          print (encodeFloat 2 6 :: Double)
          print (encodeFloat 2 6 :: Float)
          print (encodeFloat 4 (-4) :: Double)
          print (encodeFloat 4 (-4) :: Float)
          print (encodeFloat (-2) (-4) :: Double)
          print (encodeFloat (-2) (-4) :: Float)
          print (acos 2 :: Double)
          print (acos 2 :: Float)
          print $ isNaN x
          print $ isNaN y
          print $ isNaN (acos 2 :: Double)
          print $ isNaN (acos 2 :: Float)
          print $ isInfinite x
          print $ isInfinite y
          print $ isInfinite (x / 0)
          print $ isInfinite (y / 0)
          print $ isDenormalized x
          print $ isDenormalized y
          print $ isDenormalized $ (encodeFloat 1 (-127) :: Float)
          print $ isDenormalized $ (encodeFloat 1 (-1023) :: Double)
          print $ isNegativeZero (0.0 :: Float)
          print $ isNegativeZero (0.0 :: Double)
          print $ isNegativeZero ((-x) / (1 / 0))
          print $ isNegativeZero ((-y) / (1 / 0))
          print $ isIEEE x
          print $ isIEEE y
          print $ exponent x
          print $ exponent y
          print $ decodeFloat x
          print $ decodeFloat y
          print $ significand x
          print $ significand y
          print $ scaleFloat 1 x
          print $ scaleFloat 1 y
          print $ atan2 (1.0 :: Double) 1.0
          print $ atan2 (1.0 :: Float) 1.0


