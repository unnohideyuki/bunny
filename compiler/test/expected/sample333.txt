2
2
128.0
128.0
0.25
0.25
-0.125
-0.125
NaN
NaN
False
False
True
True
False
False
True
True
False
False
True
True
False
False
True
True
True
True
2
2
(7074237752028440,-51)
(13176795,-22)
0.7853981633974483
0.7853982
6.283185307179586
6.2831855
0.7853981633974483
0.7853982
